android/BOINC/app/src/main/java/edu/berkeley/boinc/client/Monitor.java

```diff
@@ -819,35 +819,48 @@ private Integer getPidForProcessName(String processName) {
     	
     	// figure out what index PID has
     	String [] headers = processLinesAr[0].split("[\\s]+");
-    	Integer PidIndex = 1;
+    	Integer PidIndex = -1;
     	for (int x = 0; x < headers.length; x++) {
     		if(headers[x].equals("PID")) {
     			PidIndex = x;
     			break;
     		}
     	}
-		if(Logging.DEBUG) Log.d(Logging.TAG,"getPidForProcessName(): PID at index: " + PidIndex + " for output: " + processLinesAr[0]);
-    	
+
+		if (PidIndex == -1)
+			return null;
+
+		if (Logging.DEBUG)
+			Log.d(Logging.TAG, "getPidForProcessName(): PID at index: " + PidIndex + " for output: " + processLinesAr[0]);
+
 		Integer pid = null;
-    	for(int y = 1; y < processLinesAr.length; y++) {
-    	    Boolean found = false;
-    	    String [] comps = processLinesAr[y].split("[\\s]+");
-    	    for(String arg: comps) {
-    	    	if(arg.equals(processName)) {
-    	    		if(Logging.DEBUG) Log.d(Logging.TAG,"getPidForProcessName(): " + processName + " found in line: " + y);
-    	    		found = true;
-                    break;
-    	    	}
-    	    }
-    	    if(found) {
-	    	    try{
-	    	    	pid = Integer.parseInt(comps[PidIndex]);
-	        	    if(Logging.ERROR) Log.d(Logging.TAG,"getPidForProcessName(): pid: " + pid); 
-	    	    }catch (NumberFormatException e) {if(Logging.ERROR) Log.e(Logging.TAG,"getPidForProcessName(): NumberFormatException for " + comps[PidIndex] + " at index: " + PidIndex);}
-    	    }
-    	}
-    	// if not happen in ps output, not running?!
-		if(pid == null) if(Logging.ERROR) Log.d(Logging.TAG,"getPidForProcessName(): " + processName + " not found in ps output!");
+		for (int y = 1; y < processLinesAr.length; y++) {
+			Boolean found = false;
+			String[] comps = processLinesAr[y].split("[\\s]+");
+			for (String arg : comps) {
+				if (arg.equals(processName)) {
+					if (Logging.DEBUG)
+						Log.d(Logging.TAG, "getPidForProcessName(): " + processName + " found in line: " + y);
+					found = true;
+					break;
+				}
+			}
+			if (found) {
+				try {
+					pid = Integer.parseInt(comps[PidIndex]);
+					if (Logging.ERROR)
+						Log.d(Logging.TAG, "getPidForProcessName(): pid: " + pid);
+				} catch (NumberFormatException e) {
+					if (Logging.ERROR)
+						Log.e(Logging.TAG, "getPidForProcessName(): NumberFormatException for " + comps[PidIndex] + " at index: " + PidIndex);
+
+					break;
```

Comment:
Since you find it confusing, just add a comment like "break out of outer for (processLinesAr.length) loop"

Reply:
Also, I think this `break` should go after the `try/catch` block as the outer for loop should be exited after the process was found, regardless of whether the PID could be extracted or not.

Reply:
True. Fixed.